src/serac/physics/utilities/functional/tensor.hpp

```diff
@@ -82,9 +82,8 @@ struct tensor;
 
 template <typename T>
 struct tensor<T> {
-  using type                    = T;
-  static constexpr int ndim     = 0;
-  static constexpr int shape[1] = {0};
+  using type                = T;
+  static constexpr int ndim = 0;
```

Comment:
I think this specialization should have a `first_dim` member too, to be consistent. In the same way, it should also probably have an `operator[]` as well!
```suggestion
  static constexpr int ndim = 0;
  static constexpr int first_dim = 1;
  SERAC_HOST_DEVICE constexpr auto& operator[](int) { return value; }
  SERAC_HOST_DEVICE constexpr auto operator[](int) const { return value; }
```